rts/Rendering/Units/UnitDrawer.cpp

```diff
@@ -547,18 +548,25 @@ void CUnitDrawerLegacy::DrawUnitIconsScreen() const
 			const float x1 = (pos.x + offset) / globalRendering->viewSizeX;
 			const float y1 = (pos.y - offset) / globalRendering->viewSizeY;
 
-			if (x1 < 0 && x0 > 1 && y0 < 0 && y1 > 1)
+			if (x1 < 0 || x0 > 1 || y0 < 0 || y1 > 1) // This was previously &&, if any of these are true, then the icon is not drawn.
 				continue; // don't try to draw when totally outside the screen
 
+			if (hasVisibleIcons == false) {
```

Comment:
This check looks redundant, just keep `hasVisibleIcons = true;` here.

Reply:
it's to avoid doing `icon->BindTexture();` for each unit though